tests/unit/test_encoding.py

```diff
@@ -0,0 +1,55 @@
+from unittest import TestCase
+
+from electionguard.encode import (
+    BYTE_ENCODING,
+    PAD_INDICATOR_SIZE,
+    TruncationError,
+    add_padding,
+    remove_padding,
+    PaddedDataSize,
+)
+
+
+class TestEncoding(TestCase):
+    """Guardian tests"""
+
+    def test_byte_padding(self) -> None:
```

Comment:
This is a very comprehensive unit test.  That said, I personally find it hard to read and hard to understand.  In my experience the most useful unit tests are small, have exactly one reason to fail, and have names with specificity.

In particular if I've broken someone else's unit test I want the name to tell me exactly what went wrong so I can fix it without having to read and understand a lot of context.

Of course the real world isn't always that easy, and this would probably be more work than worth to refactor into many smaller tests.  So let's leave it as a very comprehensive test for now, but maybe keep an eye out for opportunities to simplify in the future.